tapa/graphir_conversion/gen_rs_graphir.py

```diff
@@ -222,28 +223,61 @@ def get_submodule_inst(
     for arg in inst.args:
         port_name = arg.port
         if arg.cat == Instance.Arg.Cat.SCALAR:
-            connections.append(
-                ModuleConnection(
-                    name=port_name,
-                    hierarchical_name=HierarchicalName.get_name(port_name),
-                    expr=Expression(
-                        (Token.new_id(arg_table[inst.name][arg.name][-1].name),)
-                    ),
+            # connect port directly to literal
+            arg_name_expr = Expression.from_str_to_tokens(arg.name)
+            if arg_name_expr.is_all_literals():
+                connections.append(
+                    ModuleConnection(
+                        name=port_name,
+                        hierarchical_name=HierarchicalName.get_name(port_name),
+                        expr=arg_name_expr,
+                    )
+                )
+            else:
+                # connect port to wire
+                # get wire name from arg_table
+                conn_str = arg_table[inst.name][arg.name][-1].name
+                connections.append(
+                    ModuleConnection(
+                        name=port_name,
+                        hierarchical_name=HierarchicalName.get_name(port_name),
+                        expr=Expression.from_str_to_tokens(conn_str),
+                    )
```

Comment:
Can we do this?

```suggestion
            if not arg_name_expr.is_all_literals():
                # connect port to wire
                # get wire name from arg_table
                arg_name_expr = Expression.from_str_to_tokens(arg_table[inst.name][arg.name][-1].name)

            connections.append(
                ModuleConnection(
                    name=port_name,
                    hierarchical_name=HierarchicalName.get_name(port_name),
                    expr=arg_name_expr,
                )
```

Rationale: 1) it's easier to find the difference between the two branches, and 2) it's shorter code